ctl/src/command.rs

```diff
@@ -343,7 +338,7 @@ pub fn upgrade_main(mut channel: Channel<CommandRequest,CommandResponse>,
 
             // Reconnect to the new main
             println!("Reconnecting to new main process...");
-            let mut channel = create_channel(&config).expect("could not reconnect to the command unix socket");
+            let mut channel = create_channel(&config).context("could not reconnect to the command unix socket")?;
```

Comment:
Is there a method `with_context` that take a closure ? It will help to not allocate the String, if there is no error

Reply:
Sure !
```rust
let mut channel = create_channel(&config).with_context(|| "could not reconnect to the command unix socket")?;
```